acceptance/tests/environment/use_enc_environment_for_pluginsync.rb

```diff
@@ -11,30 +11,49 @@
 END
 on master, "chmod 755 #{testdir}/enc.rb"
 
+apply_manifest_on(master, <<-MANIFEST, :catch_failures => true)
+  File {
+    ensure => directory,
+    mode => "0770",
+    owner => #{master.puppet['user']},
+    group => #{master.puppet['group']},
+  }
+  file {
+    '#{testdir}/environments':;
+    '#{testdir}/environments/production':;
+    '#{testdir}/environments/special/':;
+    '#{testdir}/environments/special/modules':;
+    '#{testdir}/environments/special/modules/amod':;
+    '#{testdir}/environments/special/modules/amod/lib':;
+    '#{testdir}/environments/special/modules/amod/lib/puppet':;
+  }
+  file { '#{testdir}/environments/special/modules/amod/lib/puppet/foo.rb':
+    ensure => file,
+    mode => "0640",
+    content => "#special_version",
+  }
+MANIFEST
+
 master_opts = {
+  'main' => {
+    'environmentpath' => "#{testdir}/environments",
+  },
   'master' => {
     'node_terminus' => 'exec',
     'external_nodes' => "#{testdir}/enc.rb"
   },
-  'special' => {
-    'modulepath' => "#{testdir}/special"
-  }
 }
 
-on master, "mkdir -p #{testdir}/modules"
-# Create a plugin file on the master
-on master, "mkdir -p #{testdir}/special/amod/lib/puppet"
-create_remote_file(master, "#{testdir}/special/amod/lib/puppet/foo.rb", "#special_version")
-
-on master, "chown -R #{master['user']}:#{master['group']} #{testdir}"
-on master, "chmod -R g+rwX #{testdir}"
-
 with_puppet_running_on master, master_opts, testdir do
 
   agents.each do |agent|
+    agent_vardir = agent.puppet['vardir']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.